File: advent-of-code-2017/02_1.hs
import Control.Arrow ((&&&))
import Data.List.Split (splitOn)

diff :: [Int] -> Int
diff = uncurry (-) . (maximum &&& minimum)

parseRow :: String -> [Int]
parseRow = map read . splitOn "\t"

main = do
  print . sum . map (diff . parseRow) . lines =<< readFile "02_input.tsv"

File: advent-of-code-2017/02_1.clj
(require '[clojure.string :as str])

(defn diff [xs]
  (- (apply max xs) (apply min xs)))

(defn parse-row [r]
  (map read-string (str/split r #"\t")))

(defn solve []
  (->> (slurp "02_input.tsv")
       (#(str/split % #"\n"))
       (map (comp diff parse-row))
       (reduce +)))

(println (solve))
